### DISCUSSION TITLE
What is the linux kernel stack

### DISCUSSION THREAD

**Discussion by SmartSelfDev** [community/community#39770] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello everyone,
I'm taking the oswp course, Although everything is fine until now, I just have a little trouble while trying to figure out what is the Linux Wireless Stack! In fact I have a bunch of questions, I really need to answer them so to proceed to the next modules, I hope you can help me :)

So What is the Linux Wireless Stack?
What about the stack kernel driver?
And How & why wireless drivers use the stack kernel driver?
Finally, What is the main difference between ieee802.11 stack and mac802.11 stack? 
And why linux switched from ieee to mac! what was the main reason to change that stack (Stack term is pretty confusing to me :( )...

Thanks in advance :)

**Comment by airtower-luna** [comment, marked as answer]
"Linux Wireless Stack" means the parts of the kernel that work together to support wireless networking. This includes drivers, but also shared infrastructure, for example the module that lets you configure your wireless interfaces, and code that drivers can use to assemble packages. There is a whole dedicated website about it: https://wireless.wiki.kernel.org/

**Reply by SmartSelfDev** [reply]
Hi Luna,
Your answer is more than enough for me! Now I can differentiate between a lot of messes that on floating inside of my head 🥲

I thank you for you help ^^, Appreciated!
